ic-helpers/src/management/types.rs

```diff
@@ -0,0 +1,57 @@
+use candid::{CandidType, Deserialize, Principal};
+use ic_types::ic00::Payload;
+
+#[derive(CandidType, Deserialize, Debug, Clone, PartialEq, Eq, Hash)]
+pub struct Pubkey(Vec<u8>);
+
+impl std::default::Default for Pubkey {
+    fn default() -> Self {
+        Pubkey::empty()
+    }
+}
+
+impl Pubkey {
+    pub fn new(pubkey: Vec<u8>) -> Self {
+        Self(pubkey)
+    }
+
+    pub fn empty() -> Self {
+        Self(Vec::new())
+    }
+
+    pub fn as_bytes(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+#[derive(CandidType, Deserialize, Debug)]
+pub struct GetECDSAPublicKeyArgs {
+    pub canister_id: Option<Principal>,
+    pub derivation_path: Vec<Vec<u8>>,
+    pub key_id: String,
+}
+
+/// This is a structure that is only used for `get_ecdsa_pubkey` call in
+/// management canister. The only problem that it is only used by dfx in
+/// November build of ic, but as a dependency we use the more recent one,
+/// hence, to be able to deserialized the result of `get_ecdsa_pubkey`
+/// we need something that represents its response.
+#[derive(CandidType, Deserialize, Debug)]
+pub struct GetECDSAPublicKeyResponse {
+    pub public_key: Vec<u8>,
+    pub chain_code: Vec<u8>,
+}
```

Comment:
Can this be moved to management/canister.rs and be made private instead, or is another project depending on this as well from this crate?

Reply:
What do you mean exactly? Are there any fields that we want to keep private? This struct is used for deserialisation only and I used it token for extracting fields from but of course we could return something else.

Reply:
What I mean is: if this struct is only used in one place, we don't need it in types.rs (which is something I hope we can get rid of at some point) and it cane be moved to the module that is using it. Also no need to make any of it public for the same reason.

Reply:
This struct is exported and used in outside calls for deserialization